lib/cc/engine/markdownlint.rb

```diff
@@ -33,7 +35,7 @@ def include_paths
         return root unless engine_config.has_key?("include_paths")
 
         markdown_files = engine_config["include_paths"].select do |path|
-          path.end_with?(".md")
+          EXTENSIONS.include?(File.extname(path)) || path.end_with?("/")
```

Comment:
It means `mdl` will try and analyze every `*.md` file in the given directories. Which isn't quite right. So you're right, we probably need to walk the tree ourselves & get all the actual files.